presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorDistributedMetadataUpdater.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.connector;
+
+import com.facebook.presto.spi.MetadataUpdateRequest;
+import com.facebook.presto.spi.MetadataUpdateResult;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public interface ConnectorDistributedMetadataUpdater
+{
+    long addMetadataUpdateRequest(int count);
+
+    List<MetadataUpdateRequest> getPendingMetadataUpdateRequests();
+
+    void removeResultFuture(long requestId);
+
+    CompletableFuture<List<String>> getMetadataResult(long requestID);
+
+    void setMetadataUpdateResults(List<MetadataUpdateResult> results);
```

Comment:
We only need to expose interfaces that are accessed at engine level. For all that are used by connectors only, we don't need to expose them.